services/cloudfront/src/main/java/software/amazon/awssdk/services/cloudfront/CloudFrontSignedCookie.java

```diff
@@ -0,0 +1,251 @@
+    private static final String POLICY_KEY = "CloudFront-Policy";
+    private final String keyPairIdVal;
+    private final String signatureVal;
+    private final String expiresVal;
+    private final String policyVal;
+    private final boolean isCustom;
+
+    private CloudFrontSignedCookie(DefaultBuilder builder) {
+        this.keyPairIdVal = builder.keyPairIdVal;
+        this.signatureVal = builder.signatureVal;
+        this.expiresVal = builder.expiresVal;
+        this.policyVal = builder.policyVal;
+        this.isCustom = builder.isCustom;
+    }
+
+    /**
+     * Creates a builder for {@link CloudFrontSignedCookie}.
+     */
+    public static Builder builder() {
+        return new DefaultBuilder();
+    }
+
+    @Override
+    public Builder toBuilder() {
+        return new DefaultBuilder(this);
+    }
+
+    /**
+     * Returns the key pair ID key
+     */
+    public String keyPairIdKey() {
+        return KEY_PAIR_ID_KEY;
+    }
+
+    /**
+     * Returns the signature key
+     */
+    public String signatureKey() {
+        return SIGNATURE_KEY;
+    }
+
+    /**
+     * Returns the expires key
+     */
+    public String expiresKey() {
+        if (this.isCustom) {
+            throw SdkClientException.create("This is a cookie with a custom policy, use policy instead of expires.");
+        }
+        return EXPIRES_KEY;
+    }
+
+    /**
+     * Returns the policy key
+     */
+    public String policyKey() {
+        if (!this.isCustom) {
+            throw SdkClientException.create("This is a cookie with a canned policy, use expires instead of policy");
+        }
+        return POLICY_KEY;
+    }
+
+    /**
+     * Returns the key pair ID value
+     */
+    public String keyPairIdVal() {
+        return keyPairIdVal;
+    }
+
+    /**
+     * Returns the signature value
+     */
+    public String signatureVal() {
+        return signatureVal;
+    }
+
+    /**
+     * Returns the expires value
+     */
+    public String expiresVal() {
+        if (this.isCustom) {
+            throw SdkClientException.create("This is a cookie with a custom policy, use policy instead of expires.");
+        }
+        return expiresVal;
+    }
+
+    /**
+     * Returns the policy value
+     */
+    public String policyVal() {
+        if (!this.isCustom) {
+            throw SdkClientException.create("This is a cookie with a canned policy, use expires instead of policy");
+        }
+        return policyVal;
+    }
+
+    /**
+     * Returns true if this is a cookie with custom policy, else false for a cookie with canned policy
+     */
+    public boolean isCustom() {
+        return isCustom;
+    }
+
+    /**
+     * Returns the cookie header value for the specified cookie type.
+     *
+     * @param type The type of the cookie header value to retrieve
+     * @return The cookie header value to pass into an HTTP request
+     */
+    public String cookieHeaderValue(String type) {
```

Comment:
Where does this type come from and how do users know what to query for?

Reply:
Can make an enum for this